### ISSUE TITLE
How to add other payment types besides Cash?

### ISSUE THREAD

**Issue by jshrek** [opensourcepos/opensourcepos#4067]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code that [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable.

### Installation information

N/A - working with online demo

### Issue / Bug  / Question / New Feature

How to add other payment methods besides Cash?

I have only checked out the demo, and there does not appear to be an option, not even under Configuration, where additional payment types get added.

I also tried searching docs, but I am not seeing how I can add addtional payment types.

I am interested in adding Stripe (our processor) for taking payments with their card readers.

Thanks

**Comment by objecttothis** [comment]
Hello @jshrek. 3rd party integration support is in the works once 3.4.0 is released.  3.4.0 is based on CodeIgniter 4 which will allow us to create events within the framework.  Those events will be used to kick off enabled, installed and configured 3rd party integrations such as Stripe or Stripe Payment Intents.  Keep in mind that someone will need to develop the 3rd party integration for Stripe once the integration support is in place. For that you can either do it yourself, hire a freelance developer to do it or wait until someone else wants it bad enough to do it.  Keep in mind that whoever does the work, will need to submit a Pull Request and pass code reviews in order for the changes to be pulled into the master which will guarantee future updates to OSPOS don't break the integration.

**Comment by nemestrinus** [comment]
Hello, I also would be very interested in a third-party integration for the Stripe Terminal.  Now that we are up to 3.4.1, do we know if there is any work being done on this by anyone?  Unfortunately, I am a novice and would not have the know-how to attempt it myself.

Thank you to all the contributors on this great project !

**Comment by objecttothis** [comment]
> Hello, I also would be very interested in a third-party integration for the Stripe Terminal. Now that we are up to 3.4.1, do we know if there is any work being done on this by anyone? Unfortunately, I am a novice and would not have the know-how to attempt it myself.
> 
> Thank you to all the contributors on this great project !

@jshrek and @nemestrinus I have this Pull Request started which will provide the basis for third-party integrations which at this point we are calling plugins (https://github.com/opensourcepos/opensourcepos/pull/4255).  I'm assuming that Stripe Terminal provides an API for interacting with it.  If so, this would be a prime candidate for a plugin.  Plugins at it's core will use events, however some plugins may require small modifications to the code base.  For example if someone added a plugin called "FooPay", we would need to add business logic to have "FooPay" be in the dropdown to select in the sales register when FooPay is enabled.  However those changes will primarily be restricted to the views.  The rest of the business logic will be restricted to using CodeIgniter's Events framework and that's what my PR is doing.

I'm traveling during the summer, but this PR will also convert Mailchimp to this plugin architecture which can be used as an example for others wanting to create plugins.  @nemestrinus After this PR I have a few plugins that I will be adding to the codebase, but they are for things that I need in my application.  You'd be welcome to hire the work out, but #4255 needs to be completed first and the code will still need to meet our coding standards in order to be included in the code base.  You can fork the repo and do what you like, but keep in mind that future bugfixes, enhancements and such don't automatically get merged into your fork of the code.
